src/plugins/data/common/field_formats/field_format.ts

```diff
@@ -185,7 +187,7 @@ export abstract class FieldFormat {
     };
   }
 
-  private static toConvertObject(convert: FieldFormatConvertFunction): Partial<FieldFormatConvert> {
+  static toConvertObject(convert: FieldFormatConvertFunction): Partial<FieldFormatConvert> {
```

Comment:
It doesn't seem like you're importing any of the code in the data plugin, could it be that you no longer require these changes and we can just revert all the changes in this file?

Reply:
when I started to add these imports 
```
import { FieldDescriptor } from 'src/legacy/server/index_patterns/service/index_patterns_service';
import { IndexPatternsService } from 'src/legacy/server/index_patterns';
```
I started to see this error with tsc `Property 'convertObject' of exported class expression may not be private or protected.`
and I think it is all related to this issue https://github.com/Microsoft/TypeScript/issues/17293\

Reply:
Discussed with @XavierM because we use typescript's typechecking without emiting types we don't bump into this bug. However, we use typescript to emit types for all of core's source code so that we can generate documentation from it. So whenever a dependency is pulled into core, it becomes susceptible to this bug.